ambry-store/src/main/java/com/github/ambry/store/PersistentIndex.java

```diff
@@ -511,6 +516,32 @@ List<LogSegment> getLogSegments() {
     return result;
   }
 
+  /**
+   * Update the partial log segment info including the partialLogSegmentCount and wastedLogSegmentSpace
+   */
+  void updatePartialLogSegmentInfo() {
+    int partialCount = 0;
+    long wastedSpace = 0;
+    LogSegment segment = log.getFirstSegment();
+    LogSegmentName journalName = null;
+    if (journal.getCurrentNumberOfEntries() != 0) {
+      journalName = journal.getFirstOffset().getName();
+    }
+    while (segment != null) {
+      // exclude journal
+      if (journalName != null && segment.getName().equals(journalName)) {
+        break;
+      }
+      if (segment.getEndOffset() < segment.getCapacityInBytes() * 0.8) {
+        partialCount++;
+      }
+      partialCount += segment.getCapacityInBytes() - segment.getEndOffset();
```

Comment:
I think this should be 
`wastedSpace += segment.getCapacityInBytes() - segment.getEndOffset();`